modules/live-schedule-impl/src/main/java/org/opencastproject/liveschedule/impl/LiveScheduleServiceImpl.java

```diff
@@ -0,0 +1,827 @@
+        throw new LiveScheduleException("Live media package " + mp.getIdentifier() + " could not be published");
+    } catch (LiveScheduleException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new LiveScheduleException(e);
+    }
+  }
+
+  void retract(MediaPackage mp) throws LiveScheduleException {
+    try {
+      List<Job> jobs = new ArrayList<Job>();
+      Set<String> elementIds = new HashSet<String>();
+      // Remove media package from the search index
+      String mpId = mp.getIdentifier().compact();
+      logger.info("Removing LIVE media package {} from the search index", mpId);
+
+      jobs.add(searchService.delete(mpId));
+      // Retract elements
+      for (MediaPackageElement mpe : mp.getElements()) {
+        if (!MediaPackageElement.Type.Publication.equals(mpe.getElementType()))
+          elementIds.add(mpe.getIdentifier());
+      }
+      jobs.add(downloadDistributionService.retract(CHANNEL_ID, mp, elementIds));
+
+      if (!waitForStatus(jobs.toArray(new Job[jobs.size()])).isSuccess())
+        throw new LiveScheduleException("Removing live media package from search did not complete successfully");
+    } catch (LiveScheduleException e) {
+      throw e;
+    } catch (Exception e) {
+      throw new LiveScheduleException(e);
+    }
+  }
+
+  /**
+   * Retrieves the media package from the search index.
+   *
+   * @param mediaPackageId
+   *          the media package id
+   * @return the media package in the search index or null if not there
+   * @throws LiveException
+   *           if found many media packages with the same id
+   */
+  MediaPackage getMediaPackageFromSearch(String mediaPackageId) throws LiveScheduleException {
+    // Look for the media package in the search index
+    SearchQuery query = new SearchQuery().withId(mediaPackageId);
+    SearchResult result = searchService.getByQuery(query);
+    if (result.size() == 0) {
+      logger.debug("The search service doesn't know live mediapackage {}", mediaPackageId);
+      return null;
+    } else if (result.size() > 1) {
+      logger.warn("More than one live mediapackage with id {} returned from search service", mediaPackageId);
+      throw new LiveScheduleException("More than one live mediapackage with id " + mediaPackageId + " found");
+    }
+    return result.getItems()[0].getMediaPackage();
+  }
+
+  void setDuration(MediaPackage mp, DublinCoreCatalog dc) {
+    DCMIPeriod period = EncodingSchemeUtils.decodeMandatoryPeriod(dc.getFirst(DublinCore.PROPERTY_TEMPORAL));
+    long duration = period.getEnd().getTime() - period.getStart().getTime();
+    mp.setDuration(duration);
+    logger.debug("Live media package {} has start {} and duration {}", mp.getIdentifier(), mp.getDate(),
+            mp.getDuration());
+  }
+
+  void addLiveTracks(MediaPackage mp, String caName) throws LiveScheduleException {
+    String mpId = mp.getIdentifier().compact();
+    try {
+      // MHPearl specific: if capture agent registered the properties:
```

Comment:
Remove "MHPearl specific", i.e. just

      // If capture agent registered the properties: